extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -181,88 +180,82 @@ export function combineInventoryUnits(instances) {
   };
 }
 
-/* eslint-disable jsdoc/require-param */
 /**
- * Array of functions used to combine block level request parameters for SRA
- * request.
- * @private @const {!Array<!function(!Array<AmpAdNetworkDoubleclickImpl>):?Object<string,string>>}
+ * Given array of instances, having previous called populateAdUrlState, join
+ * block state into SRA parameter.
+ * @enum {function(!Array<AmpAdNetworkDoubleclickImpl>):?Object<string,string>}
+ * @visibleForTesting
  */
-const BLOCK_SRA_COMBINERS_ = [
-  combineInventoryUnits,
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate cookie opt out.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return instance.jsonTargeting_ &&
-        instance.jsonTargeting_['cookieOptOut'] ? {'co': '1'} : null;
-  }),
-  instances => {
-    return {'adks': instances.map(instance => instance.adKey_).join()};
-  },
-  instances => {
-    return {'prev_iu_szs': instances.map(instance =>
-      instance.parameterSize_).join()};
-  },
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate TFCD.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return instance.jsonTargeting_ && instance.jsonTargeting_[TFCD] ?
-      {'tfcd': instance.jsonTargeting_[TFCD]} : null;
-  }),
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate manual experiment.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return isInManualExperiment(instance.element) ? {'adtest': 'on'} : null;
-  }),
-  instances => {
+export const SRA_JOINER = {
+  IU: combineInventoryUnits,
+  COOKIE_OPT_OUT: impls => getFirstInstanceValue_(impls, impl =>
+    impl.jsonTargeting_ &&
+        impl.jsonTargeting_['cookieOptOut'] ? {'co': '1'} : null),
```

Comment:
On further reading of this, I strongly suggest you break all of these combiners into a separate module, and then make each combiner its own function with jsdoc. As it is right now, it does not lend itself to widespread understanding

Reply:
On the upshot, if all our code is like this, we'd never need to use an obfuscator :)

Reply:
https://en.wikipedia.org/wiki/List_of_burn_centers_in_the_United_States

Reply:
The number of times I've had to remind Keith that our code gets automatically minified after he writes it.....